src/openms/include/OpenMS/ANALYSIS/QUANTITATION/PeptideAndProteinQuant.h

```diff
@@ -203,9 +217,63 @@ namespace OpenMS
     /**
          @brief Gather quantitative information from a feature.
 
-         Store quantitative information from @p feature in member @p pep_quant_, based on the peptide annotation in @p hit. If @p hit is empty ("ambiguous/no annotation"), nothing is stored.
+         Store quantitative information from @p feature in member @p pep_quant_, based on the peptide annotation in @p hit. 
+         @p fraction, use 0 for first fraction (or if no fractionation was performed)
+         @p sample, use 0 for first sample, 1 for second, ... 
+         If @p hit is empty ("ambiguous/no annotation"), nothing is stored.
     */
-    void quantifyFeature_(const FeatureHandle& feature, const PeptideHit& hit);
+    void quantifyFeature_(const FeatureHandle& feature, 
+      size_t fraction, 
+      size_t sample, 
+      const PeptideHit& hit);
+
+    /*
+     *   @brief Determine fraction and charge state of a peptide with the highest
+     *   number of abundances.
+     *   @return true if at least one abundance was found, false otherwise
+     */ 
+    bool getBest_(
+      const std::map<Int, std::map<Int, SampleAbundances>> & peptide_abundances, 
+      std::pair<size_t, size_t> & best)
+    {
+      size_t best_n_quant(0);
+      double best_abundance(0);
+      best = std::make_pair(0,0);
+
+      for (auto & fa : peptide_abundances)  // for all fractions 
+      {
+        for (auto & ca : fa.second) // for all charge states
+        {
+          const Int & fraction = fa.first;
+          const Int & charge = ca.first;
+
+          double current_abundance(0);
+          for (auto & sa : ca.second) // loop over abundances
+          {
+            const double & sample_abundance = sa.second;
+            current_abundance += sample_abundance;
+          }
+
+          if (current_abundance <= 0) { continue; }
+
+          const size_t current_n_quant = ca.second.size();
+          if (current_n_quant > best_n_quant)
+          {           
+            best_abundance = current_abundance;
+            best_n_quant = current_n_quant;
+            best = std::make_pair(fraction, charge);
+          }
+          else if (current_n_quant == best_n_quant 
+            && current_abundance > best_abundance)  // resolve tie by abundance
+          {
+            best_abundance = current_abundance;
+            best = std::make_pair(fraction, charge);
```

Comment:
why make the pair? you dont use it anywhere ...

Reply:
I think it is a non-const input parameter to the function. Used in Cpp line 211 then.
@timosachsenberg can you comment?

Reply:
Added doxygen for this